### PR TITLE
Fixes from first round of auditor feedback

### PR DESCRIPTION
**Pull request by mootz12**

### CODE REVIEW

File: contracts/votes/src/balance.rs [hunk 1]
```diff
@@ -57,64 +57,68 @@ pub fn mint_balance(e: &Env, to: &Address, amount: i128) {
 /// ### Panics
 /// This function panics if the balance is less than the amount to remove.
 pub fn burn_balance(e: &Env, from: &Address, amount: i128) {
-    let balance = storage::get_balance(e, from);
-    if balance < amount {
-        panic_with_error!(e, TokenVotesError::BalanceError);
-    }
+    if amount > 0 {
```
Comment (markuspluna) [inline contracts/votes/src/balance.rs:60]: Why don't we just panic if not over 0

File: contracts/votes/src/balance.rs [hunk 2]
```diff
@@ -57,64 +57,68 @@ pub fn mint_balance(e: &Env, to: &Address, amount: i128) {
...
 
-    #[cfg(feature = "bonding")]
-    {
-        let total_supply_checkpoint = storage::get_total_supply(e);
-        let (_, supply) = total_supply_checkpoint.to_checkpoint_data();
-        emissions::update_emissions(e, supply, from, from_balance);
-        emissions::update_emissions(e, supply, to, to_balance);
-    }
+        let to_balance = storage::get_balance(e, to);
+        storage::set_balance(e, to, &(to_balance + amount));
```
Comment (markuspluna) [inline contracts/votes/src/balance.rs:105]: wrong order no? will overwrite from

Comment (markuspluna) [reply contracts/votes/src/balance.rs:105]: if same address

Comment (markuspluna) [reply contracts/votes/src/balance.rs:105]: nvm i dumb

**Review by markuspluna — APPROVED** [review]
few comments but looks like it solves vulns fine

File: contracts/votes/src/balance.rs (hunk 1 above)
Comment (mootz12) [reply contracts/votes/src/balance.rs:60]: we panic if its less than 0, current the implemented native token contract does not panic if called with a transfer of 0, so this allows 0 to be input but won't actually do anything.

File: contracts/tests/tests/votes/test_get_past.rs [hunk 3]
```diff
@@ -236,40 +236,44 @@ fn test_past_checkpoints_get_pruned() {
...
-    let start_vote_3 = cur_ledger + 10 * ONE_DAY_LEDGERS - 1;
+    let start_vote_1 = cur_ledger + 3 * ONE_DAY_LEDGERS - 1;
+    let start_vote_2 = cur_ledger + 15 * ONE_DAY_LEDGERS - 1;
+    let start_vote_3 = cur_ledger + 16 * ONE_DAY_LEDGERS - 1;
     votes_client.set_vote_sequence(&start_vote_0);
     votes_client.set_vote_sequence(&start_vote_1);
     votes_client.set_vote_sequence(&start_vote_2);
     votes_client.set_vote_sequence(&start_vote_3);
 
     // Time = 10 days ago
```
Comment (Ryang-21) [inline contracts/tests/tests/votes/test_get_past.rs:247]: (nit picky) These I assume are not needed anymore

**Review by Ryang-21 — APPROVED** [review]

**Pull request merged**